src/spec/sync/product-sync.spec.coffee

```diff
@@ -226,6 +226,71 @@ describe 'ProductSync', ->
         version: oldProduct.version
       expect(update).toEqual expected_update
 
+    it 'should create `changePrice` action if new price has id', ->
+      oldProduct =
+        id: '123'
+        version: 1
+        masterVariant:
+          id: 1
+          sku: 'v1'
+          prices: [
+            {id: 'p-1', value: {currencyCode: 'EUR', centAmount: 100}},
+            {id: 'p-2', value: {currencyCode: 'EUR', centAmount: 1000}},
+          ]
+        variants: [
+          {
+            id: 2
+            prices: [
+              {id: 'p-7', value: {currencyCode: 'EUR', centAmount: 2000}},
+              {id: 'p-8', value: {currencyCode: 'EUR', centAmount: 2100}, country: 'FR'},
+            ]
+          }
+          {
+            id: 3
+            prices: [
+              {id: 'p-10', value: {currencyCode: 'EUR', centAmount: 5889}, country: 'DE'},
+            ]
+          }
+        ]
+
+      newProduct =
+        id: '123'
+        masterVariant:
+          sku: 'v1'
+          prices: [
+            {id: 'p-1', value: {currencyCode: 'GBP', centAmount: 555}},
+            {id: 'p-2', value: {currencyCode: 'GBP', centAmount: 245}},
```

Comment:
I am curious what is the business use case that new/updated product variant (supplied by external system) is aware of internal IDs generated by CTP API like price.id?

Reply:
You do not have to rely on price IDs. You can identify/match prices by its scope: currency, country, customer group and channel.

Reply:
@butenkor There are instances where Prices are exported, modified and imported (with their IDs) back to the platform. Currently, the old prices were removed and the new prices were added. This changes the `priceId` of the price and that's what this issue addresses

The scope fields are optional, and can even be shared by multiple prices in one variant

Reply:
Use case is valid but would fail if you export from production CTP project and want to import into staged project as IDs there will be different. Matching by scope will cover both your case as the one described by me without relying on generated internal IDs.  Even if scope field are optional you can not have prices in API with duplicate scope. So each scope combination (no scope defined is a scope also) is like unique ID which can be used for matching prices on variant level.